cocos/renderer/CCGroupCommand.cpp

```diff
@@ -54,27 +55,31 @@ int GroupCommandManager::getGroupID()
         int groupID = *_unusedIDs.rbegin();
         _unusedIDs.pop_back();
         _groupMapping[groupID] = true;
+        _usedIDs.insert(groupID);
         return groupID;
     }
 
     //Create new ID
 //    int newID = _groupMapping.size();
     int newID = Director::getInstance()->getRenderer()->createRenderQueue();
     _groupMapping[newID] = true;
-
+    _usedIDs.insert(newID);
     return newID;
 }
 
 void GroupCommandManager::releaseGroupID(int groupID)
 {
-    _groupMapping[groupID] = false;
-    _unusedIDs.push_back(groupID);
+    if (groupID > 0) {
+        _groupMapping[groupID] = false;
+        _usedIDs.erase(groupID);
+        _unusedIDs.push_back(groupID);
+    }
 }
 
 GroupCommand::GroupCommand()
 {
     _type = RenderCommand::Type::GROUP_COMMAND;
-    _renderQueueID = Director::getInstance()->getRenderer()->getGroupCommandManager()->getGroupID();
```

Comment:
what happened if GroupCommand::init() is not invoked?

Reply:
If GroupCommand::init() is not invoked implies that command not receive a _renderQueueID and game crashes :/ ... But i think that is not correct to a GroupCommand receive a _renderQueueID in GroupCommand::GroupCommand(). Suppose a scene with 250 ui::Layout with clipping disabled, ui::Layout has a instance of a GroupCommand, but it is used only when Layout clipping is enabled, the old code will create 250 RenderQueue and it will never be deleted! In ever frame the code calls .sort() and .clear() over all RenderGroups forever...

Reply:
May be should add assert in `GroupCommand::getRenderQueueID()` to ask to invoke `GroupCommand::init()` if it is not invoked.

Reply:
If not, then it is hard to detect the crash error.